File: bonus_content/tik_tok_effect_for_children/tik_tok_effect_src/Emoji Puzzles 4 Kids/Graph/ScriptTemplate/Sequence/CGSequence.lua
local CGSequence = CGSequence or {}
CGSequence.__index = CGSequence

function CGSequence.new()
    local self = setmetatable({}, CGSequence)
    self.nexts = {}
    return self
end

function CGSequence:setNext(index, func)
    self.nexts[index] = func
end

function CGSequence:execute(index)
    if self.nexts[0] then
        self.nexts[0]()
    end
    if self.__contextStack ~= nil and #self.__contextStack > 0 then
    	if self.__contextStack[#self.__contextStack][1] == "return" then
    		return
   		end
    end
    if self.nexts[1] then
        self.nexts[1]()
    end
    if self.__contextStack ~= nil and #self.__contextStack > 0 then
    	if self.__contextStack[#self.__contextStack][1] == "return" then
    		return
   		end
    end
    if self.nexts[2] then
        self.nexts[2]()
    end
end

return CGSequence

File: bonus_content/tik_tok_effect_for_children/tik_tok_effect_src/Emoji Puzzles 4 Kids/Graph/ScriptTemplate/Sequence/CGSequence.js
/**
 * @file CGSequence.js
 * @author liujiacheng
 * @date 2021/8/23
 * @brief CGSequence.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGSequence extends BaseNode {
  constructor() {
    super();
  }

  execute(index) {
    for (let k in this.nexts) {
      if (this.nexts[k]) {
        this.nexts[k]();
      }
    }
  }
}

exports.CGSequence = CGSequence;
